### PR TITLE
Create packages namespaces via helm

### PR DESCRIPTION
**Pull request by ewollesen**
Previously, the helm chart created a specific namespace per cluster:

Self-managed clusters: "eksa-packages"
Workload clusters:     "eksa-packages-$CLUSTER_NAME"

The PBC controller was also checking for, and creating if necessary, "eksa-packages-$CLUSTER_NAME", which would only trigger for self-managed clusters, as the helm chart was creating it for workload clusters.

With this change, both "eksa-packages" and "eksa-packages-$CLUSTER_NAME" are created via the helm chart. The code to create them from the PBC controller is removed. This should simplify the entire process.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: charts/eks-anywhere-packages/templates/namespace.yaml [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
 apiVersion: v1
 kind: Namespace
 metadata:
   annotations:
     "helm.sh/resource-policy": keep
   creationTimestamp: null
-  {{- if not .Values.workloadOnly }}
   name: eksa-packages
-  {{- else }}
+spec: {}
```
Comment (lewisdiamond) [inline charts/eks-anywhere-packages/templates/namespace.yaml:9]: I'd remove the spec and status if empty.

Comment (ewollesen) [reply charts/eks-anywhere-packages/templates/namespace.yaml]: Done. I didn't add those, but sure, can't hurt to remove them. Just extra noise in the review.

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Review by vincentni — APPROVED** [review]

**Review by vincentni — APPROVED** [review]
/lgtm
/approve

File: charts/eks-anywhere-packages/templates/namespace.yaml [hunk 2]
```diff
@@ -1,10 +1,17 @@
-{{ $namespace := ternary "eksa-packages" (printf "%s-%s" "eksa-packages" .Values.clusterName) .Values.workloadOnly }}
-{{- if not (lookup "v1" "Namespace" "" $namespace ) }}
+{{- if not .Values.workloadOnly }}
```
Comment (lewisdiamond) [inline charts/eks-anywhere-packages/templates/namespace.yaml:1]: I don't think this change is necessary considering the latest version.

**Comment by ewollesen** [comment]
After some discussion with @lewisdiamond we think we've come to a good arrangement that will work now, and will ease the transition to packages as a package that is to come shortly. There were some subtle edge cases he's seen that we can avoid, at the minor cost of a little extra complexity in this PR.

I'll be pushing a new commit shortly.

**Pull request merged**